frc46_token/src/token/mod.rs

```diff
@@ -159,9 +167,9 @@ where
 
         // Increase the balance of the actor and increase total supply
         let result = self.transaction(|state, bs| {
-            let balance = state.change_balance_by(&bs, owner_id, amount)?;
-            let supply = state.change_supply_by(amount)?;
-            Ok(MintReturn { balance, supply: supply.clone(), recipient_data: RawBytes::default() })
+            state.change_balance_by(&bs, owner_id, amount)?;
+            state.change_supply_by(amount)?;
```

Comment:
Discussion: in most cases, when there is no recursion, the value returned here will be the right one. It would be nice to avoid traversing the balance table again to look up the value when that happens. So storing it in the intermediate for that case could help. That then raises the question of flowing information through to the token so that it knows whether it needs to look up a new balance. Maybe an internal dirty flag or mutation count?

I think this is a nuanced enough issue that you should write an issue and defer it to a new PR. Let's get it correct now, and then optimise.

Reply:
Made issue https://github.com/helix-onchain/filecoin/issues/110 for it